src/components/GatsbyImage.tsx

```diff
@@ -0,0 +1,21 @@
+import { chakra, type ChakraComponent } from "@chakra-ui/react"
+import {
+  GatsbyImage as Image,
+  type GatsbyImageProps,
+} from "gatsby-plugin-image"
+
+export type GatsbyImageType = ChakraComponent<"img", GatsbyImageProps>
+
+/**
+ * Custom version of `GatsbyImage` optimized for Chakra
+ */
+const GatsbyImage: GatsbyImageType = chakra(Image, {
+  shouldForwardProp: (prop) => {
+    // Forward props that only GatsbyImage should be applying, else warnings may throw
+    return ["image", "loading", "objectFit", "alt", "objectPosition"].includes(
```

Comment:
I'm afk at the time of this comment, but this list is still missing props that need to be forwarded.

Reply:
Full list of props https://www.gatsbyjs.com/docs/reference/built-in-components/gatsby-plugin-image/